specification/batch/Azure.Batch/routes.tsp

```diff
@@ -0,0 +1,2544 @@
+background.
+""")
+  @route("/jobs/{jobId}/tasks/{taskId}")
+  @delete
+  deleteTask is RpcOperation<
+    BatchClientRequestHeaders &
+      BatchMatchHeaders & {
+        @doc("The ID of the Job from which to delete the Task.")
+        @path
+        jobId: string;
+
+        @doc("The ID of the Task to delete.")
+        @path
+        taskId: string;
+      },
+    BatchClientResponseHeaders & {
+      @doc("A process exit code.")
+      @statusCode
+      code: "200";
+    },
+    {},
+    BatchError
+  >;
+
+  @summary("Gets information about the specified Task.")
+  @doc("""
+For multi-instance Tasks, information such as affinityId, executionInfo and
+nodeInfo refer to the primary Task. Use the list subtasks API to retrieve
+information about subtasks.
+""")
+  @route("/jobs/{jobId}/tasks/{taskId}")
+  @get
+  getTask is RpcOperation<
+    BatchClientRequestHeaders &
+      BatchMatchHeaders & {
+        @doc("The ID of the Job that contains the Task.")
+        @path
+        jobId: string;
+
+        @doc("The ID of the Task to get information about.")
+        @path
+        taskId: string;
+
+        @doc("An OData $select clause.")
+        @query({
+          format: "csv",
+        })
+        $select?: string[];
+
+        @doc("An OData $expand clause.")
+        @query({
+          format: "csv",
+        })
+        $expand?: string[];
+      },
+    BatchResponseHeaders &
+      BatchTask & {
+        @header("DataServiceId")
+        @doc("The OData ID of the resource to which the request applied.")
+        DataServiceId: string;
+      },
+    {},
+    BatchError
+  >;
+
+  @doc("Updates the properties of the specified Task.")
+  @route("/jobs/{jobId}/tasks/{taskId}")
+  @put
+  replaceTask is RpcOperation<
+    BatchClientRequestHeaders &
+      BatchMatchHeaders & {
+        @doc("Type of content")
+        @header("content-type")
+        contentType: "application/json; odata=minimalmetadata";
+
+        @doc("The ID of the Job containing the Task.")
+        @path
+        jobId: string;
+
+        @doc("The ID of the Task to update.")
+        @path
+        taskId: string;
+
+        @doc("The Task to update.")
+        @body
+        body: BatchTask;
+      },
+    BatchResponseHeaders & {
+      @doc("A process exit code.")
+      @statusCode
+      code: "200";
+
+      @header("DataServiceId")
+      @doc("The OData ID of the resource to which the request applied.")
+      DataServiceId: string;
+    },
+    {},
+    BatchError
+  >;
+
+  #suppress "@azure-tools/typespec-azure-core/use-standard-names" "GET operations that return single objects should start with 'get'"
+  @summary("""
+Lists all of the subtasks that are associated with the specified multi-instance
+Task.
+""")
+  @doc("If the Task is not a multi-instance Task then this returns an empty collection.")
+  @route("/jobs/{jobId}/tasks/{taskId}/subtasksinfo")
+  @get
+  listSubTasks is RpcOperation<
```

Comment:
I'm just wondering if this is a paging operation?

Reply:
No, it's not pageable in the current design.

Reply:
After discussion, we do want Cx to have a consistent experieces with our `list` operations, and to make it as a paging operation as well. But right now the services won't reply `odata.nextLink` for this endpoint, is it OK we add it in our Typespec?

Reply:
I think it is okay to make it paginable if your service will change it to paging operation in near future. @weidongxu-microsoft Any suggestions?

Reply:
It is fine that service does not return "nextLink" for the response in runtime.

As long as there is a chance that the response would be pageable (aka service would return "nextLink" in future), it is highly recommended that service add the "nextLink" (it's an optional property) to TypeSpec/Swagger now.

This way, if user uses REST API directly, they would know (from Swagger/TypeSpec) that they would need to have the logic to handle "nextLink", when it present in future.